### PR TITLE
fix reduction repeated axis error

### PR DESCRIPTION
**Pull request by mattjj**
Before PR:

```python
In [1]: import jax.numpy as jnp
In [2]: jnp.sum(jnp.arange(3), (0, 0))
Out[2]: DeviceArray(9, dtype=int32)
```

Seems bad!

### CODE REVIEW

**Review by froystig — APPROVED** [review]

**Comment by sanjoy** [comment]
I'm also going to fix XLA's `ShapeInference` to outlaw repeated reduction indices.

**Comment by mattjj** [comment]
Thanks @sanjoy !
